### DISCUSSION TITLE
How to use TypeScript types as type arguments?

### DISCUSSION THREAD

**Discussion by krutoo** [webpack/webpack#16906] [Q&A] [answered]
Hi, in my project i use `Configuration` property types as type arguments.

When i trying to build project with declarations its fails with error like:
```
Exported variable 'factories' has or is using name 'Output' 
from external module ".../node_modules/webpack/types" but cannot be named.
```

Here is an example repo of problem:
https://github.com/krutoo/try-webpack-types

This problem is repeated with several other types, such as `externals`, `externalsPresets` and `optimization`

How can i import for example `output` configuration property type to solve this issue?

**Comment by alexander-akait** [comment, marked as answer]
Can you try this:
```
import type { Configuration } from "webpack";

type MyOutput = Configuration["output"] | undefined;

function createFactory<T>(): () => T {
  return () => null as unknown as T;
}

export const factories = {
  output: createFactory<MyOutput>(),
};
```
?

**Reply by krutoo** [reply]
Looks like its working, thanks
